public/app/core/components/AppChrome/AppChrome.tsx

```diff
@@ -181,8 +181,9 @@ const getStyles = (theme: GrafanaTheme2, searchBarHidden: boolean) => {
       },
       {
         position: 'fixed',
-        height: `calc(100% - ${searchBarHidden || isSingleTopNav ? TOP_BAR_LEVEL_HEIGHT : TOP_BAR_LEVEL_HEIGHT * 2}px)`,
+        height: `calc(100% - ${isSingleTopNav ? 0 : searchBarHidden ? TOP_BAR_LEVEL_HEIGHT : TOP_BAR_LEVEL_HEIGHT * 2}px)`,
         zIndex: 2,
+        top: isSingleTopNav ? 0 : searchBarHidden ? TOP_BAR_LEVEL_HEIGHT : TOP_BAR_LEVEL_HEIGHT * 2,
```

Comment:
maybe just add a new block to the css declaration then we can avoid the nested ternaries:
```
    dockedMegaMenu: css(
      {
        background: theme.colors.background.primary,
        borderRight: `1px solid ${theme.colors.border.weak}`,
        display: 'none',
        width: MENU_WIDTH,

        [theme.breakpoints.up('xl')]: {
          display: 'block',
        },
      },
      {
        position: 'fixed',
        height: `calc(100% - ${searchBarHidden ? TOP_BAR_LEVEL_HEIGHT : TOP_BAR_LEVEL_HEIGHT * 2}px)`,
        zIndex: 2,
      },
      isSingleTopNav && {
        height: '100%',
        top: 0,
      }
    ),
```